dexter/src/main/java/com/karumi/dexter/listener/threaddecorator/MultiplePermissionListenerThreadDecorator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2015 Karumi.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.karumi.dexter.listener.threaddecorator;
+
+import com.karumi.dexter.MultiplePermissionsReport;
+import com.karumi.dexter.PermissionToken;
+import com.karumi.dexter.listener.PermissionRequest;
+import com.karumi.dexter.listener.multi.MultiplePermissionsListener;
+import java.util.List;
+
+/**
+ * Decorator to execute the permission updates on a given thread specification
+ */
+public class MultiplePermissionListenerThreadDecorator implements MultiplePermissionsListener {
```

Comment:
I'd say the same about the ``ThreadSpec`` interface, it doesn't make sense to expose that part of the library. As I see it we are adding a new abstraction for no real benefit. Users of the library can actually wrap their listeners to execute its callbacks using any android/custom abstraction, with that interface we would be forcing them to understand yet another concept.